plugin/gcp_config.go

```diff
@@ -0,0 +1,43 @@
+package gcpauth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-gcp-common/gcputil"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/iam/v1"
+)
+
+// gcpConfig contains all config required for the GCP backend.
+type gcpConfig struct {
```

Comment:
This struct was pulled out of `path_config.go`:
- `formatAndMarshalCredentials` function was unchanged (other than renaming the receiver from `config` to `c` for consistency).
- `Update` got new lines for the `iam_alias` and `gce_alias` fields.
- `getIamAlias` and `getGCEAlias` are copied from `aliasing.go` and made to be a part of the `gcpConfig` struct.